src/lib/operators/jit_operator/operators/jit_compute.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "jit_abstract_operator.hpp"
+
+namespace opossum {
+
+/* JitExpression represents a SQL expression - this includes arithmetic and logical expressions as well as comparisons.
+ * JitExpressions work on JitTupleValues and are structured as a binary tree. All leafs of that tree reference a tuple
+ * value in the JitRuntimeContext and are of type ExpressionType::Column - independent of whether these values actually
+ * came from a column, are literal values or placeholders.
+ * Each JitExpressions can compute its value and stores it in its assigned result JitTupleValue. JitExpressions are also
+ * able to compute the data type of the expression they represent.
+ *
+ * Using AbstractExpression as a base class for JitExpressions seems like a logical choice. However, AbstractExpression
+ * adds a lot of bloat during code specialization. We thus decided against deriving from it here.
+ */
+class JitExpression {
```

Comment:
Having this in `jit_compute.hpp` is surprising. In most cases, we have one class per file.

Reply:
Ping?